src/frequenz/channels/util/_select.py

```diff
@@ -106,6 +109,13 @@ def __del__(self) -> None:
         for task in self._pending:
             task.cancel()
 
+    async def stop(self) -> None:
+        """Stop the `Select` instance and cleanup any pending tasks."""
+        for task in self._pending:
+            task.cancel()
+        await asyncio.gather(*self._pending, return_exceptions=True)
```

Comment:
Shouldn't this throw exceptions instead of returning them? Or this is just because you assume the only exception that could be captured is the `CancelledError` and this is why you are ignoring them? If so it might be worth a clarifying comment

Same for the other PR adding a stop().